### PR TITLE
Add logging on slow UFS calls

### PR DESCRIPTION
**Pull request by apc999**
Add information on UFS call logging:
(1) time elapsed for each UFS call
(2) add a new flag to control log UFS calls that takes time longer than defined threshold.

### CODE REVIEW

File: core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java [hunk 1]
```diff
@@ -52,7 +55,8 @@
  */
 public class UnderFileSystemWithLogging implements UnderFileSystem {
   private static final Logger LOG = LoggerFactory.getLogger(UnderFileSystemWithLogging.class);
-
+  private static final long THRESHOLD = new InstancedConfiguration(ConfigurationUtils.defaults())
```
Comment (gpang) [inline core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java:58]: Should this just use the `mConf` to get the property?

File: core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java [hunk 2]
```diff
@@ -1203,14 +1207,27 @@ public String toString() {
...
+        LOG.warn("{}({}) returned OK in {} ms (>={} ms)", methodName,
+            callable, durationMs, THRESHOLD);
+      }
       return ret;
     } catch (IOException e) {
+      durationMs = System.currentTimeMillis() - startMs;
       MetricsSystem.counter(getQualifiedFailureMetricName(methodName)).inc();
-      LOG.debug("Exit (Error): {}: {}, Error={}", methodName, callable, e.getMessage());
+      LOG.debug("Exit (Error): {}({}) in {} ms, Error={}",
+          methodName, callable, e.getMessage(), durationMs);
```
Comment (gpang) [inline core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java:1226]: I think the arguments are in the wrong order. `e.getMessage()` and `durationMs` should be switched.

File: core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java [hunk 3]
```diff
@@ -1203,14 +1207,27 @@ public String toString() {
...
+      if (durationMs >= THRESHOLD) {
+        LOG.warn("{}({}) returned {} in {} ms (>={} ms)", methodName,
```
Comment (gpang) [inline core/common/src/main/java/alluxio/underfs/UnderFileSystemWithLogging.java:1228]: NIT: maybe enclose the error message in "" or ``?

**Review by gpang — CHANGES_REQUESTED** [review]
@apc999 Thanks!

What about any data read/write operations? This won't be able to log any of those UFS calls.

**Review by gpang — APPROVED** [review]
LGTM

**Comment by apc999** [comment]
alluxio-bot, merge this please

**Pull request merged**